### PR TITLE
Add --random-ports to let docker do the port selection

### PR DESCRIPTION
**Pull request by rmohr**
User "cli port [k8s|ocp|ssh|registry]" to get the assigned ports.

Signed-off-by: Roman Mohr <rmohr@redhat.com>

### CODE REVIEW

**Comment by cynepco3hahue** [comment]
lgtm, I think we can to change providers to use random ports and fetch the relevant port(k8s, openshift and registry on runtime)

**Comment by rmohr** [comment]
> lgtm, I think we can to change providers to use random ports and fetch the relevant port(k8s, openshift and registry on runtime)

Exactly. PR is here: https://github.com/kubevirt/kubevirt/pull/822/

File: k8s-1.9.3/scripts/node01.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ kind: MasterConfiguration
 apiServerExtraArgs:
   runtime-config: admissionregistration.k8s.io/v1alpha1
 token: abcdef.1234567890123456
-kubernetes-version: v1.9.3
+kubernetesVersion: v1.9.3
```
Comment (rmohr) [inline k8s-1.9.3/scripts/node01.sh:11]: The k8s-1.9.3 extra layer is there because of this line change (not a cluster rebuild, just copied the file over).

**Review by cynepco3hahue — APPROVED** [review]
lgtm, I just have a single question
Does docker-1.12.6-71.git3e8e77d.el7.centos use overlay2 or devicemapper storage driver?

**Comment by cynepco3hahue** [comment]
because under docker documentation they write explicitly `Do not attempt to use overlay2 with kernel versions older than 3.10.0-693.`

**Comment by cynepco3hahue** [comment]
By the way, all these battles with windmills just show that use of pre-baked images it is a very good idea :+1:

**Comment by rmohr** [comment]
> because under docker documentation they write explicitly Do not attempt to use overlay2 with kernel versions older than 3.10.0-693.

The newer centos version used docker 1.13.1 with overlayfs2. Now we use 1.12.6 with devicemapper and loopback inside the vm. We can improve that later. Also it can be that docker 1.13.1 works fine and we would just have to disable the metdown and spectre fixes inside our vms to solve all our issues. We will see.

**Comment by cynepco3hahue** [comment]
Cool that what I wanted to know :smile:

**Pull request merged**